tree/30_generic_methods/group_absent.cf

```diff
@@ -0,0 +1,58 @@
+#####################################################################################
+# Copyright 2015 Normation SAS
+#####################################################################################
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, Version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#####################################################################################
+
+# @name Group absent
+# @description Create a group
+# @agent_version >=3.6
+#
+# @parameter group Group name
+
+# @class_prefix group_absent
+# @class_parameter login
+# This bundle will define a class group_absent_${login}_{kept,repaired,not_ok,ok,reached}
+
+bundle agent group_absent(group)
+{
+  vars:
+      "class_prefix"        string => canonify("group_absent_${group}");
+
+      "group_id"            int => getgid("${group}");
```

Comment:
groupexists would ease the method
https://docs.cfengine.com/docs/3.5/reference-functions-groupexists.html